src/Sylius/Bundle/CoreBundle/Test/Services/PaymentMethodFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Test\Services;
+
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class PaymentMethodFactory implements PaymentMethodFactoryInterface
```

Comment:
I would put this in following namespace: ``Sylius\Component\Core\Test\Factory\PaymentMethodFactory`` and remove ``createFromArray`` method and add something like ``createForTest($name)`` and that's it. Parameters like description can be random and gateway figured out from the map just like you do right now.